### DISCUSSION TITLE
Backup producing small file size

### DISCUSSION THREAD

**Discussion by Progress3168** [teslamate-org/teslamate#3879] [Q&A]
I'm hoping someone can point me in the right direction for solving a backup issue I'm having.

I've had Teslamate running on a Pi 4 for over two months so have a decent amount of data now logged.  I configured backups as per instructions but noticed that the backup file sizes weren't increasing.  I decided to leave it for a while to get a longer term view but two months later the backup file size still states 545 bytes.

To eliminate any issues with the script I'm running, I'm now trying to troubleshoot using the backup command only, but the same file size occurs.

root@PiPlex:/home/pi/teslamate# ls -l
-rw-r--r-- 1 pi   pi   1235 Mar 29 23:33 docker-compose.yml
-rw-r--r-- 1 root root  545 May  3 19:43 teslamate.bck

I've found similar issues listed in other discussions but couldn't find anything stating exactly what the issue was and how it was resolved.  I've been trying for the past week to get to the bottom of the issue and work through suggestions from other discussions but now wonder if it's something obvious that I've become blinded to!  

I'm using the following:
sudo /usr/bin/docker-compose exec -T database pg_dump -U teslamate teslamate > teslamate.bck

Any pointers or suggestions of what else to check would be great.

**Comment by cwanja** [comment]
What happens with a non-root user? What happens without `sudo`

**Reply by Progress3168** [reply]
non-root user - Permission denied

**Reply by cwanja** [reply]
What does adding a verbose flag to the backup command provide?

**Reply by cwanja** [reply]
Also if you open the file, what are the contents?

**Reply by Progress3168** [reply]
> What does adding a verbose flag to the backup command provide?

pi@PiPlex:~/teslamate $ sudo /usr/bin/docker-compose --verbose exec -T database pg_dump -U teslamate teslamate > teslamate.bck
compose.config.config.find: Using configuration files: ./docker-compose.yml
compose.cli.docker_client.get_client: docker-compose version 1.29.2, build unknown
docker-py version: 5.0.3
CPython version: 3.11.2
OpenSSL version: OpenSSL 3.0.11 19 Sep 2023
compose.cli.docker_client.get_client: Docker base_url: http+docker://localhost
compose.cli.docker_client.get_client: Docker version: Platform={'Name': 'Docker Engine - Community'}, Components=[{'Name': 'Engine', 'Version': '26.0.0', 'Details': {'ApiVersion': '1.45', 'Arch': 'arm64', 'BuildTime': '2024-03-20T15:18:02.000000000+00:00', 'Experimental': 'false', 'GitCommit': '8b79278', 'GoVersion': 'go1.21.8', 'KernelVersion': '6.6.20+rpt-rpi-v8', 'MinAPIVersion': '1.24', 'Os': 'linux'}}, {'Name': 'containerd', 'Version': '1.6.28', 'Details': {'GitCommit': 'ae07eda36dd25f8a1b98dfbf587313b99c0190bb'}}, {'Name': 'runc', 'Version': '1.1.12', 'Details': {'GitCommit': 'v1.1.12-0-g51d5e94'}}, {'Name': 'docker-init', 'Version': '0.19.0', 'Details': {'GitCommit': 'de40ad0'}}], Version=26.0.0, ApiVersion=1.45, MinAPIVersion=1.24, GitCommit=8b79278, GoVersion=go1.21.8, Os=linux, Arch=arm64, KernelVersion=6.6.20+rpt-rpi-v8, BuildTime=2024-03-20T15:18:02.000000000+00:00
compose.cli.verbose_proxy.proxy_callable: docker inspect_network <- ('teslamate_default')
compose.cli.verbose_proxy.proxy_callable: docker inspect_network -> {'Attachable': False,
 'ConfigFrom': {'Network': ''},
 'ConfigOnly': False,
 'Containers': {'2ddada9a6bcc835105f1b58f2f56aefe811bfe05ca4599a28083361401bc0884': {'EndpointID': '0d3bfc3f60c51769b143cf20be355f6d2e3a3326c0c30830a168437f2343ed36',
'IPv4Address': '172.18.0.2/16',
'IPv6Address': '',
'MacAddress': '02:42:ac:12:00:02',
'Name': 'teslamate-database-1'}},
 'Created': '2024-03-29T23:59:51.365009549Z',
 'Driver': 'bridge',
...
compose.cli.verbose_proxy.proxy_callable: docker inspect_volume <- ('teslamate_teslamate-db')
compose.cli.verbose_proxy.proxy_callable: docker inspect_volume -> {'CreatedAt': '2024-02-27T22:20:01Z',
 'Driver': 'local',
 'Labels': {'com.docker.compose.project': 'teslamate',
            'com.docker.compose.version': '1.29.2',
            'com.docker.compose.volume': 'teslamate-db'},
 'Mountpoint': '/var/lib/docker/volumes/teslamate_teslamate-db/_data',
 'Name': 'teslamate_teslamate-db',
 'Options': None,
 'Scope': 'local'}
compose.cli.verbose_proxy.proxy_callable: docker inspect_volume <- ('teslamate_teslamate-grafana-data')
compose.cli.verbose_proxy.proxy_callable: docker inspect_volume -> {'CreatedAt': '2024-02-27T22:20:01Z',
 'Driver': 'local',
 'Labels': {'com.docker.compose.project': 'teslamate',
            'com.docker.compose.version': '1.29.2',
            'com.docker.compose.volume': 'teslamate-grafana-data'},
 'Mountpoint': '/var/lib/docker/volumes/teslamate_teslamate-grafana-data/_data',
 'Name': 'teslamate_teslamate-grafana-data',
 'Options': None,
 'Scope': 'local'}
compose.cli.verbose_proxy.proxy_callable: docker containers <- (all=False, filters={'label': ['com.docker.compose.project=teslamate', 'com.docker.compose.service=database', 'com.docker.compose.oneoff=False', 'com.docker.compose.container-number=1']})
compose.cli.verbose_proxy.proxy_callable: docker containers -> (list with 1 items)
compose.cli.main.call_docker: /usr/bin/docker exec --interactive 2ddada9a6bcc835105f1b58f2f56aefe811bfe05ca4599a28083361401bc0884 pg_dump -U teslamate teslamate

**Reply by Progress3168** [reply]
> Also if you open the file, what are the contents?

--
-- PostgreSQL database dump
--

-- Dumped from database version 14.11 (Debian 14.11-1.pgdg120+2)
-- Dumped by pg_dump version 14.11 (Debian 14.11-1.pgdg120+2)

SET statement_timeout = 0;
SET lock_timeout = 0;
SET idle_in_transaction_session_timeout = 0;
SET client_encoding = 'UTF8';
SET standard_conforming_strings = on;
SELECT pg_catalog.set_config('search_path', '', false);
SET check_function_bodies = false;
SET xmloption = content;
SET client_min_messages = warning;
SET row_security = off;

--
-- PostgreSQL database dump complete
--
